src/onboarding/containers/CloudLoginPage.tsx

```diff
@@ -59,7 +59,19 @@ export const CloudLoginPage: FC = () => {
         fetch('/api/env/quartz-login-url')
           .then(async response => {
             const quartzUrl = await response.text()
-            const pathname = window.location.pathname ?? ''
+            const redirectTo = new URLSearchParams(window.location.search).get(
+              'redirectTo'
+            )
+            let pathname
+            try {
+              // if there is a redirectTo, parse the pathname from redirect location
+              const redirectToUrl = new URL(redirectTo).pathname
```

Comment:
the `URL()` throws an error if the argument is invalid URL, so we need to handle that.